Tests/BSONTests/BSONCorpusTests.swift

```diff
@@ -129,97 +160,96 @@ final class BSONCorpusTests: BSONTestCase {
+                        if let rEJ = test.relaxedExtJSON {
+                            expect(try relaxedEncoder.encode(docFromDB))
+                                .to(cleanEqual(rEJ), description: test.description)
+                        }
+                    }
 
-                    //     // native_to_bson( json_to_native(dEJ) ) = cB (unless lossy)
-                    //     if !lossy {
-                    //         expect(try Document(fromJSON: dEJ).rawBSON).to(equal(cBData))
-                    //     }
-                    // }
+                    // for dEJ input (if it exists):
+                    if let dEJ = test.degenerateExtJSON, let dEJData = dEJ.data(using: .utf8) {
+                        // native_to_canonical_extended_json( json_to_native(dEJ) ) = cEJ
+                        expect(try canonicalEncoder.encode(try decoder.decode(BSON.self, from: dEJData)))
+                            .to(cleanEqual(test.canonicalExtJSON), description: test.description)
+                        // native_to_bson( json_to_native(dEJ) ) = cB (unless lossy)
+                        if !lossy {
+                            try expect(try decoder.decode(BSONDocument.self, from: dEJData))
+                                .to(sortedEqual(BSONDocument(fromBSON: cBData)), description: test.description)
+                        }
+                    }
 
                     // for rEJ input (if it exists):
-                    // if let rEJ = test.relaxedExtJSON {
-                    //     // native_to_relaxed_extended_json( json_to_native(rEJ) ) = rEJ
-                    //     expect(try Document(fromJSON: rEJ).extendedJSON).to(cleanEqual(rEJ))
-                    // }
+                    if let rEJ = test.relaxedExtJSON, let rEJData = rEJ.data(using: .utf8) {
+                        // native_to_relaxed_extended_json( json_to_native(rEJ) ) = rEJ
+                        expect(try relaxedEncoder.encode(try decoder.decode(BSON.self, from: rEJData)))
+                            .to(cleanEqual(rEJ), description: test.description)
+                    }
                 }
             }
 
             if let parseErrorTests = testFile.parseErrors {
                 continue // TODO: EXT JSON support required
                 for test in parseErrorTests {
+                    guard !shouldSkip(testFile.description, test.description) else {
+                        continue
+                    }
                     let description = "\(testFile.description)-\(test.description)"
-
+                    guard let testData = Data(hexString: test.string) else {
+                        XCTFail("Unable to interpret canonical_bson as Data")
+                        return
+                    }
                     switch BSONType(rawValue: UInt8(testFile.bsonType.dropFirst(2), radix: 16)!)! {
                     case .invalid: // "top level document" uses 0x00 for the bson type
                         _ = ()
```

Comment:
can we remove the `_ = ()` no-op here and below.

Reply:
yup